src/ruby/puppet-server-lib/puppet/server/master.rb

```diff
@@ -57,6 +60,53 @@ def handleRequest(request)
         response["X-Puppet-Version"])
   end
 
+  def getClassInfoForAllEnvironments()
+    # The clear_environment_settings(env) ensures that the calls which happen
+    # later on to enumerate all of the manifests for each environment are
+    # using the latest environment.conf settings for each environment.  Without
+    # this call, cached (and, therefore, possibly stale) environment.conf
+    # settings would be used.  We want to return the latest data for any calls
+    # made to this method.
+    @env_loader.list.each do |env|
+      Puppet.settings.clear_environment_settings(env)
+    end
+
+    environments =
+      Hash[@env_loader.list.collect do |env|
+       [env.name, self.class.getManifests(env)]
+      end]
```

Comment:
So, @cprice404, assuming that plumbing the new API up from core Ruby Puppet and integrating with Puppet Server will take some time, what do you suggest we do with this PR in the meantime?  Commit it as is, with a separate ticket to replace this work with the new API as it becomes available later, or just shelve it and come back to it later.  I suppose over the next few days that we may mostly be consumed with work outside of the "NC Class Refresh" effort, but this will quickly become a blocker for making further progress on that epic as we do have cycles to put toward it.